app/models/user.rb

```diff
@@ -375,6 +375,15 @@ def identity
     @identity ||= UserIdentity.find(uuid)
   end
 
+  def onboarding
+    if Flipper.enabled?(:veteran_onboarding_beta_flow, self) && user_account_uuid.present?
```

Comment:
`user_account` and `user_account_uuid` will be present for every user, so you don't need the check here.

As an additional note, I know before you were checking to make sure `ICN` existed. Is onboarding for all veterans or just verified ones? There's probably slightly different functionality you want if this is just for verified veterans

Reply:
This needs to be for onboarding veterans only.  I will re-add the check for icn.

Reply:
I still think you might want a different set of attributes to check here. Is this intended for only users who are new to VA.gov?

Reply:
For the first iterations of this project, we will have a set of test users/beta users who we apply the feature toggle to.  Those folks will see the new onboarding process.  Once we get a base set of features, we will more critically examine what will determine an "onboarding veteran"

Reply:
To get the information you mentioned 'verified within 30 days', that is available, it's on the `user.user_verification.verified_at`. This describes when we (va.gov) first saw the user as `verified`

Reply:
Right, yeah currently we have only the one check: whether or not the user has dismissed the "onboarding" screen.

I'm sure you can think of a scenario where you are new to a site, you get a modal or some kind of overlay ("Tip of the day", etc).  You can check a box which is like, "don't show this again".

The current check we have in place, `display_onboarding_flow` is like that checkbox.  I will have a followup PR with a new Rails controller, to allow the front-end client to POST and modify that value.

In the future, that Rails endpoint will also be the place where the client will save a veteran's place in a user flow of some sort (maybe a wizard? maybe a set of preferences? that has not entirely been decided yet).  Right now we just want to allow the user to dismiss the onboarding flow (or re-enable it).

Reply:
sounds good

I just want to confirm because I wasn't exactly clear from this convo, you want this onboarding to only be shown to verified users, right? That's what the `icn.present?` check does but I wanted to confirm (for clarity, a verified user is someone who has confirmed their identity by doing something like uploading their driver's license. Where an unverified user could have an account, but might not have input any personal data yet)